### PR TITLE
[Feature] Add groups parameter to Conv2d in ResidualBlockNoBN

### LINKED ISSUE #1512: [Feature] Add groups parameter to Conv2d in ResidualBlockNoBN
### What's the feature?

I'm working on implement acceleration and compression for Super Resolution model to deploy on small device.
I have to use group convolution using conv2d groups parameter. But in mmediting ResidualBlockNoBN has no groups parameter.
So I think groups parameter has to be add to ResidualBlockNoBN for who need group convolution.

### Any other context?

Add groups parameter to Conv2d in ResidualBlockNoBN

### PR DESCRIPTION
**Pull request by gihwan-kim**
**Add groups parameter to Conv2d in ResidualBlockNoBN**

Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

I'm working on implement acceleration and compression for Super Resolution model to deploy on small device.  
I have to use group convolution using conv2d groups parameter. But in mmediting ResidualBlockNoBN has no groups parameter.  
So I think groups parameter has to be add to ResidualBlockNoBN for who need group convolution.

## Modification

Please briefly describe what modification is made in this PR.

Add groups parameter to Conv2d in ResidualBlockNoBN.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmediting/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmediting/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmedit/models/common/sr_backbone_utils.py [hunk 1]
```diff
@@ -56,11 +56,11 @@ class ResidualBlockNoBN(nn.Module):
             Default: 1.0.
     """
 
-    def __init__(self, mid_channels=64, res_scale=1.0):
+    def __init__(self, mid_channels=64, res_scale=1.0, groups=1):
         super().__init__()
         self.res_scale = res_scale
-        self.conv1 = nn.Conv2d(mid_channels, mid_channels, 3, 1, 1, bias=True)
-        self.conv2 = nn.Conv2d(mid_channels, mid_channels, 3, 1, 1, bias=True)
+        self.conv1 = nn.Conv2d(mid_channels, mid_channels, 3, 1, 1, bias=True, groups=1)
```
Comment (plyfager) [inline mmedit/models/common/sr_backbone_utils.py:62]:
```suggestion
        self.conv1 = nn.Conv2d(mid_channels, mid_channels, 3, 1, 1, bias=True, groups= groups)
```

File: mmedit/models/common/sr_backbone_utils.py [hunk 2]
```diff
@@ -56,11 +56,11 @@ class ResidualBlockNoBN(nn.Module):
...
+        self.conv2 = nn.Conv2d(mid_channels, mid_channels, 3, 1, 1, bias=True, groups=1)
```
Comment (plyfager) [inline mmedit/models/common/sr_backbone_utils.py:63]:
```suggestion
        self.conv2 = nn.Conv2d(mid_channels, mid_channels, 3, 1, 1, bias=True, groups= groups)
```

**Comment by plyfager** [comment]
Sorry for responding later. You may fix the comment so I can merge this PR. 
circleci failed for this PR. You may pip install ``pre-commit`` and run ``pre-commit run --all-files`` to fix this error.

**Comment by plyfager** [comment]
This may also be useful for branch ``dev-1.x``, we will appreciate it a lot if you can also make a pr to fix this in `dev-1.x`.

**Comment by plyfager** [comment]
Hi @gihwan-kim , we'll just handle this PR since hasn't receive further response.

**Comment by plyfager** [comment]
Hi @gihwan-kim , Could you sign the CLA so we can merge this PR?

**Review by plyfager — APPROVED** [review]

**Review by plyfager — APPROVED** [review]

**Comment by OpenMMLab-Team** [comment]
Hi @gihwan-kim！First of all, we want to express our gratitude for your significant PR in the MMEditing . Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA

If you have WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution！❤

**Pull request merged**